flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/stream/FsStreamingSinkITCaseBase.scala

```diff
@@ -139,7 +208,8 @@ abstract class FsStreamingSinkITCaseBase extends StreamingTestBase {
   }
 }
 
-class FiniteTestSource(elements: Iterable[Row]) extends SourceFunction[Row] with CheckpointListener{
+class FiniteTestSource(elements: Iterable[Row], fun: Row => Long) extends SourceFunction[Row]
```

Comment:
hint: a meaningful name like `fun` -> `watermarkGenerator`

Reply:
great, thank you.